compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala

```diff
@@ -51,6 +52,7 @@ class PruneErasedDefs extends MiniPhase with SymTransformer { thisTransform =>
     else cpy.ValDef(tree)(rhs = trivialErasedTree(tree.rhs))
 
   override def transformDefDef(tree: DefDef)(using Context): Tree =
+    RefChecks.checkNoInlineAnnoClasses(tree)
```

Comment:
Oh, I think my previous comment was not clear. I meant why adding the check in the `PruneErasedDefs` and not in `RefChecks` ?

Reply:
Ah, I see! I asked @nicolasstucki and he mentioned that if it happened at a later stage then some inlined definitions wouldn't be available, but I'm not 100% sure about that.